File: algorithms/1_easy/C/01784_check_if_binary_string_has_at_most_one_segment_of_ones/solution.py
class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        n = len(s)
        count = 0

        for i in range(n):
            if s[i] == '1' and (i == 0 or s[i - 1] == '0'):
                count += 1

                if count > 1:
                    return False

        return True

File: algorithms/1_easy/C/01784_check_if_binary_string_has_at_most_one_segment_of_ones/solution.rs
impl Solution {
    pub fn check_ones_segment(s: String) -> bool {
        let bytes = s.as_bytes();
        let len = bytes.len();
        let mut count = 0_i32;

        for i in 0..len {
            if bytes[i] == b'1' && (i == 0 || bytes[i - 1] == b'0') {
                count += 1;

                if count > 1 {
                    return false;
                }
            }
        }

        true
    }
}
